### DISCUSSION TITLE
Typescript errors during build

### DISCUSSION THREAD

**Discussion by Kay2dan** [blitz-js/blitz#1514] [Q&A]
This is really odd and I wonder if someone else has faced a similar issue.

I have been fixing the issues typescript fails the build on. One I am stuck with is:

```
export default async function logout(
  _: null = null,
  ctx: { session?: SessionContext } = {}
): Promise<void> {
  return await ctx.session!.revoke();
}
````

Usage:
````
<a
onClick={async (): Promise<void> => {
  await logout();
  ...
}}>
Logout
</a>

````

Typescript keeps screaming: `Type error: Expected 1-2 arguments, but got 0.` This is most frustrating and with most of the files where I have mutation funcs, this is coming up and I am having to resort to adding `// @ts-nocheck` at the top of the files to make typescript ignore them.

How are you all dealing with these error?

**Comment by flybayer** [comment]
What blitz version are you on?

That logout code is from an old version.

Here's the latest code:
- https://github.com/blitz-js/blitz/blob/canary/packages/generator/templates/app/app/auth/mutations/logout.ts
- https://github.com/blitz-js/blitz/blob/canary/packages/generator/templates/app/app/pages/index.tsx#L14

**Comment by Kay2dan** [comment]
Ah, I have keeping up with the upgrades, however I have been skipping the files with conflicts. I will go back & check, however, I do not think this is the cause of Typescript issue :(

WIll update after the update. Thank you as always for the help and proactiveness. Cheers.
